Clarifies Helm version in docs

samgd opened the pull request:
Helm 2 is now legacy as Helm 3 has been released. This PR clarifies that the Infrastructure documentation refers to Helm 2.

SamG97 commented:
Is it worth adding an explicit comment about Helm 3 not currently being supported but it's something planned for the future?

samgd commented:
Added the clarification about being in legacy mode.

Explicitly not added discussion about Helm 3 for two reasons: 1. We haven't discussed what this means and whether we want to adopt Helm 3 or use this as an opportunity to switch to another option/system. 2. Future roadmap/plans should probably be in a self-contained place that can easily be seen and updated.
